### PR TITLE
[FLINK-36323][StateBackend] Remove deprecated MemoryStateBackend, FsStateBackend and RocksDBStateBackend

### PR DESCRIPTION
**Pull request by xing1mo**
## What is the purpose of the change
After [FLINK-19463](https://issues.apache.org/jira/browse/FLINK-19463), the checkpoint storage has been split from the state backends. The deprecated implementation MemoryStateBackend and RocksDBStateBackend are replaced by HashMapStateBackend and EmbeddedRocksDBStateBackend. Those two state backends should be removed in Flink 2.0.

## Brief change log
  - *Remove MemoryStateBackend, FsStateBackend, RocksDBStateBackend. And change related test.

## Verifying this change
This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: yes
  - The serializers: no 
  - The runtime per-record code paths (performance sensitive): no / don't know
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no 
  - The S3 file system connector: no 

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/operator/CEPOperatorTest.java [hunk 1]
```diff
@@ -444,7 +447,7 @@ public void testKeyedCEPOperatorNFAUpdateWithRocksDB() throws Exception {
             operator = CepOperatorTestUtilities.getKeyedCepOperator(true, new SimpleNFAFactory());
             harness = CepOperatorTestUtilities.getCepTestHarness(operator);
 
-            rocksDBStateBackend = new RocksDBStateBackend(new MemoryStateBackend());
+            rocksDBStateBackend = new EmbeddedRocksDBStateBackend(TernaryBoolean.TRUE);
```
Comment (Zakelly) [inline flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/operator/CEPOperatorTest.java:450]: Should we use the constructor without any parameter, `new EmbeddedRocksDBStateBackend();` ?

Same for any other places

File: flink-runtime/src/main/java/org/apache/flink/runtime/state/StateBackendLoader.java [hunk 2]
```diff
@@ -90,9 +82,7 @@ public class StateBackendLoader {
      * factory is instantiated (via its zero-argument constructor) and its {@link
      * StateBackendFactory#createFromConfig(ReadableConfig, ClassLoader)} method is called.
      *
-     * <p>Recognized shortcut names are '{@value StateBackendLoader#HASHMAP_STATE_BACKEND_NAME}',
-     * '{@value StateBackendLoader#ROCKSDB_STATE_BACKEND_NAME}' and '{@value
-     * StateBackendLoader#MEMORY_STATE_BACKEND_NAME}' (Deprecated)
+     * <p>Recognized shortcut names are '{@value StateBackendLoader#HASHMAP_STATE_BACKEND_NAME}'
```
Comment (Zakelly) [inline flink-runtime/src/main/java/org/apache/flink/runtime/state/StateBackendLoader.java:85]: Don't remove `StateBackendLoader#ROCKSDB_STATE_BACKEND_NAME` here

File: flink-runtime/src/test/java/org/apache/flink/runtime/state/StateBackendLoadingTest.java [hunk 3]
```diff
@@ -86,153 +83,49 @@ void testApplicationDefinedHasPrecedence() throws Exception {
...
      * configuration.
      */
     @Test
-    void testLoadMemoryStateWithParameters() throws Exception {
-        final String checkpointDir = new Path(TempDirUtils.newFolder(tmp).toURI()).toString();
-        final String savepointDir = new Path(TempDirUtils.newFolder(tmp).toURI()).toString();
-        final Path expectedCheckpointPath = new Path(checkpointDir);
-        final Path expectedSavepointPath = new Path(savepointDir);
-
+    void testLoadHashMapStateBackendWithParameters() throws Exception {
```
Comment (Zakelly) [inline flink-runtime/src/test/java/org/apache/flink/runtime/state/StateBackendLoadingTest.java:102]: IIUC, you should remove this Test as it test the Hashmap with parameters, but actually there is no parameter set.

File: flink-runtime/src/test/java/org/apache/flink/runtime/state/FileStateBackendMigrationTest.java [hunk 4]
```diff
@@ -27,14 +28,20 @@
 
 /**
  * Tests for the keyed state backend and operator state backend, as created by the {@link
- * FsStateBackend}.
+ * HashMapStateBackend}.
  */
 @ExtendWith(NoOpTestExtension.class)
-public class FileStateBackendMigrationTest extends StateBackendMigrationTestBase<FsStateBackend> {
+public class FileStateBackendMigrationTest
```
Comment (Zakelly) [inline flink-runtime/src/test/java/org/apache/flink/runtime/state/FileStateBackendMigrationTest.java:34]: Rename this test?

File: flink-tests/src/test/java/org/apache/flink/test/state/operator/restore/StreamOperatorSnapshotRestoreTest.java [hunk 5]
```diff
@@ -156,16 +155,16 @@ private void testOperatorStatesSnapshotRestoreInternal(final int mode) throws Ex
         // -------------------------------------------------------------------------- snapshot
 
         StateBackend stateBackend;
-        FsStateBackend fsstateBackend = createStateBackendInternal();
+        HashMapStateBackend hashMapStateBackend = createStateBackendInternal();
         switch (stateBackendEnum) {
             case FILE:
-                stateBackend = fsstateBackend;
+                stateBackend = hashMapStateBackend;
```
Comment (Zakelly) [inline flink-tests/src/test/java/org/apache/flink/test/state/operator/restore/StreamOperatorSnapshotRestoreTest.java:161]: Should add ` testHarness.setCheckpointStorage(xxx);` in line 212

**Review by Zakelly — COMMENTED** [review]
Thanks for the PR! Left some some comments here

**Comment by xing1mo** [comment]
@flinkbot run azure

**Comment by xing1mo** [comment]
@flinkbot run azure

**Comment by xing1mo** [comment]
@flinkbot run azure

**Comment by xing1mo** [comment]
@flinkbot run azure

**Comment by xing1mo** [comment]
@flinkbot run azure

**Comment by xing1mo** [comment]
The CI not run, I new a pr(https://github.com/apache/flink/pull/25421)

**Comment by Zakelly** [comment]
@flinkbot run azure

**Pull request closed without merging**